AppController/djinn.rb

```diff
@@ -4744,74 +4747,75 @@ def check_running_apps()
           to_start << app
           no_appservers << app
         elsif not MonitInterface.is_running?("#{app}-#{port}")
-          Djinn.log_warn("Appserver for #{app} at port #{port} is not running.")
-          to_end << "#{app}:#{port}"
+          Djinn.log_debug("Didn't find the AppServer for #{app} at port #{port}.")
+          if @last_decision[app]
+            if Time.now.to_i - @last_decision[app] > APP_UPLOAD_TIMEOUT * RETRIES
+              Djinn.log_warn("AppServer for #{app} at port #{port} is not running.")
+              to_end << "#{app}:#{port}"
+            end
+          end
         else
           my_apps << "#{app}:#{port}"
-          no_appservers.delete(app)
         end
       }
     }
-    Djinn.log_debug("Running appservers on this node: #{my_apps}.") if !my_apps.empty?
-
-    # We want to give priorities to applications that have no appserver
-    # running.
-    to_start.each{ |app|
-      no_appservers.delete(app) if to_start.count(app) != no_appservers.count(app)
+    # Let's make sure we have the proper list of apps with no currently
+    # running appservers.
+    my_apps.each { |appserver|
+      app, port = appserver.split(":")
```

Comment:
Well, that's how is done here. app_info_map store the appengine as ip:port so I kept the same convention. So that deserialization is needed to restore port and host. Do we have a better way or suggestion?